extension/test/integration/make-payment.handler.spec.js

```diff
@@ -0,0 +1,74 @@
+const { expect } = require('chai')
+
+const ctpClientBuilder = require('../../src/ctp/ctp-client')
+const iTSetUp = require('./integration-test-set-up')
+const c = require('../../src/config/constants')
+
+describe('::makePayment::', () => {
+  let ctpClient
+
+  before(async () => {
+    ctpClient = ctpClientBuilder.get()
+    await iTSetUp.initServerAndExtension(ctpClient)
+  })
+
+  afterEach(async () => {
+    await iTSetUp.cleanupResources()
+  })
+
+  it('given a payment ' +
+    'when makePayment custom field is set and response from Adyen is Authorised, ' +
+    'then should set custom field makePaymentResponse, ' +
+    'should add transaction, ' +
+    'should add interface interactions', async () => {
+    const makePaymentRequestDraft = {
+      amount: {
+        currency: 'USD',
+        value: 1000
+      },
+      reference: 'testReference',
+      paymentMethod: {
+        type: 'scheme',
+        encryptedCardNumber: 'test_4111111111111111',
+        encryptedExpiryMonth: 'test_03',
+        encryptedExpiryYear: 'test_2030',
+        encryptedSecurityCode: 'test_737'
+      },
+      returnUrl: 'https://your-company.com/'
+    }
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: c.CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: 'type',
+          key: c.CTP_PAYMENT_CUSTOM_TYPE_KEY
+        },
+        fields: {
+          makePaymentRequest: JSON.stringify(makePaymentRequestDraft)
+        }
+      }
+    }
+
+    const { statusCode, body: payment } = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    expect(statusCode).to.equal(201)
+
+    const { makePaymentResponse } = payment.custom.fields
```

Comment:
it would be nice to expect the response does not have the additionalData, which could have sensitive data in it.

Reply:
It also could be nice to ensure the expected result code from that response.

Reply:
> it would be nice to expect the response does not have the additionalData, which could have sensitive data in it.

I remember that we said somewhere that additionalData could have sensitive data, but do you know where this statement came from? I also see there could be useful info for the frontend/shopper, e.g.:
```
{
    "additionalData": {
        "cvcResult": "1 Matches",
        "avsResult": "4 AVS not supported for this card type",
        "authorisationMid": "1000",
        "cardHolderName": "Checkout Shopper PlaceHolder",
        "acquirerAccountCode": "TestPmmAcquirerAccount"
    },
    "pspReference": "853587634082404F",
    "refusalReason": "FRAUD-CANCELLED",
    "resultCode": "Refused",
    "refusalReasonCode": "22",
    "merchantReference": "testReference"
}
```
So I'm thinking maybe we could save it to interface interaction and not to the custom field?

Reply:
> It also could be nice to ensure the expected result code from that response.

https://github.com/commercetools/commercetools-adyen-integration/pull/257/commits/0f3b916e18f26f2536d9e7a29ab9de5afcc5f659

Reply:
No, we should not save it, a customer is requested to remove the field from payment object because it includes sensitive data in it, the field is customizable, it could include like the customer card details also we are not storing API-keys. Check: https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/v52/payments__resParam_additionalData

Reply:
> there could be useful info for the frontend/shopper
in which flow it could be useful?

Reply:
> > there could be useful info for the frontend/shopper
> 
> in which flow it could be useful?

For example it could contain AVS result, which is the result of the risk check as I see here: https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/v52/payments__resParam_additionalData-ResponseAdditionalDataCommon-avsResult
If we don't save this info anywhere, so no one will know why the card was discarded, and I guess Adyen provides this info so the shop can make further decisions. How do you know that we should remove it? Did we discuss it somewhere? I also remember that we should remove it, but now I don't know what is the reason.

Reply:
please check the SUPPORT-6715

Reply:
Thanks, that was the place I was looking for :D Let me clarify with Adyen as well about this.

Reply:
According to Adyen, shop could use this info for further logic, so I don't think we should strip it away. Maybe in the support ticket it was meant only the additionalData in request or just some particular fields? Because if you strip away this info, then what's the use and why Adyen would return them?

Reply:
I am not sure if this PR is the place to discuss/change that feature. Additional data is customizable on the Adyen merchant center, so I assume the data in here will not be required for any payment methods. Could you please move this discussion to another issue so we could focus it later and maybe discuss it in detail with the Adyen, WDYT?

Reply:
I agree: https://github.com/commercetools/commercetools-adyen-integration/issues/260